flake.nix

```diff
@@ -14,16 +14,34 @@
 
   outputs = { self, flake-compat, nixpkgs, utils }: utils.lib.eachDefaultSystem (system: let
     pkgs = import nixpkgs { inherit system; };
-  in {
-    packages.nixpkgs-hammer = pkgs.runCommand "nixpkgs-hammer" {
+  in rec {
+
+    packages.ast-checks = pkgs.rustPlatform.buildRustPackage {
+      name = "ast-checks";
+      cargoSha256 = "130qcmra644xmixmyrskwbx40r9fbbzql3gc35y3vq4bz9jgg81n";
+      src = ./ast-checks;
+    };
```

Comment:
Idk how @jtojnar feels, but you may want to use naersk to avoid the annoyance of manually updating this sha every time you change deps.

Reply:
I wasn't familiar with naersk. I'll give it a try.